src/ray/raylet/scheduling_resources.h

```diff
@@ -8,25 +8,64 @@
 
 #include "ray/raylet/format/node_manager_generated.h"
 
-#define EPSILON 0.00001
-
 namespace ray {
 
 namespace raylet {
 
 const std::string kCPU_ResourceLabel = "CPU";
 
-/// \brief Test if the quantity is within epsilon bounds of 0.
-///
-/// \param quantity: Quantity to check
-/// \return True if  -epsilon <= Quantity <= epsilon, False otherwise.
-bool EqualsZeroEpsilon(double quantity);
+/// \class FractionalResourceQuantity
+/// \brief Converts the resource quantities to an internal representation to
+/// avoid machine precision errors.
+class FractionalResourceQuantity {
+ public:
+  /// \brief Construct a FractionalResourceQuantity representing zero
+  /// resources.
+  FractionalResourceQuantity();
```

Comment:
Why do we need this constructor? Can we get rid of it?

Reply:
We have to have an empty constructor because when we add values to `resource_capacity_` in `ResourceSets`, `std::unordered_map` makes use of the default constructor of zero arguments. 

https://github.com/ray-project/ray/blob/1592ead002086768b8ec6ea609bd9a625a1dc08d/src/ray/raylet/scheduling_resources.cc#L99-L104